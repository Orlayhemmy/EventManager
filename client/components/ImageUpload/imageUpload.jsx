import React from 'react';

/**
 * @description imageUpload component
 */
export default class ImageUpload extends React.Component {
  /**
   * @memberof imageUpload
   * @method render
   * @description it renders the component
   * @returns the HTML of imageUpload
   */
  render() {
    return (
      <div>
        {this.props.uploadedImage === '' || !this.props.uploadedImage ? (
          <div className="imageUpload">
            <label for="imageInput">
              <div className="imageUpload">
                <img
                  src={
                    this.props.path === '/add-center'
                      ? './images/centerholder.jpg'
                      : './images/imageholder.jpg'
                  }
                  className="img-fluid dropzone"
                />
              </div>
            </label>
            <input
              type="file"
              id="imageInput"
              onChange={this.props.showImage}
            />
          </div>
        ) : (
          <div className="imageUpload">
            <label for="imageInput">
              <img
                src={this.props.uploadedImage}
                className="img-fluid dropzone"
              />
            </label>
            <input
              type="file"
              id="imageInput"
              onChange={this.props.showImage}
            />
          </div>
        )}
      </div>
    );
  }
}
