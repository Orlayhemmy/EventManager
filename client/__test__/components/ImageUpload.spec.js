import React from 'react';
import ImageUpload from '../../components/ImageUpload/ImageUpload';

describe('Image upload', () => {
  const props = {
    showImage: jest.fn(),
    uploadedImage: 'www.webimage.com/images',
    path: '/add-center'
  };
  const wrapper = shallow(<ImageUpload {...props} />);

  it('should render the component without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('change the image placeholder with the default want', () => {
    wrapper.setProps({
      ...wrapper.props,
      uploadedImage: '',
    });
    expect(wrapper.find('.imageUpload').length).toEqual(2);
  });
});
