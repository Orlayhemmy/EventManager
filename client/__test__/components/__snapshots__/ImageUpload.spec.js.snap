// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image upload should render the component without errors 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImageUpload
    path="/add-center"
    showImage={[MockFunction]}
    uploadedImage="www.webimage.com/images"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="imageUpload"
      >
        <label
          htmlFor="imageInput"
        >
          <img
            className="img-fluid dropzone"
            src="www.webimage.com/images"
          />
        </label>
        <input
          id="imageInput"
          onChange={[MockFunction]}
          type="file"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="imageInput"
          >
            <img
              className="img-fluid dropzone"
              src="www.webimage.com/images"
            />
          </label>,
          <input
            id="imageInput"
            onChange={[MockFunction]}
            type="file"
          />,
        ],
        "className": "imageUpload",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="img-fluid dropzone"
              src="www.webimage.com/images"
            />,
            "htmlFor": "imageInput",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "img-fluid dropzone",
              "src": "www.webimage.com/images",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "imageInput",
            "onChange": [MockFunction],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="imageUpload"
        >
          <label
            htmlFor="imageInput"
          >
            <img
              className="img-fluid dropzone"
              src="www.webimage.com/images"
            />
          </label>
          <input
            id="imageInput"
            onChange={[MockFunction]}
            type="file"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="imageInput"
            >
              <img
                className="img-fluid dropzone"
                src="www.webimage.com/images"
              />
            </label>,
            <input
              id="imageInput"
              onChange={[MockFunction]}
              type="file"
            />,
          ],
          "className": "imageUpload",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="img-fluid dropzone"
                src="www.webimage.com/images"
              />,
              "htmlFor": "imageInput",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "img-fluid dropzone",
                "src": "www.webimage.com/images",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "imageInput",
              "onChange": [MockFunction],
              "type": "file",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
